packages/medusa-js/src/jwt-token-manager.ts

```diff
@@ -0,0 +1,30 @@
+/**
+ * `JwtTokenManager` holds JWT tokens in state.
+ */
+class JwtTokenManager {
+  private adminJwt: string | null = null;
+  private storeJwt: string | null = null;
+
+  /**
+   * Set a publishable api key to be sent with each request.
+   */
+  public registerJwt(token: string, domain: "admin" | "store") {
+    if (domain === "admin") {
+      this.adminJwt = token;
+    } else {
+      this.storeJwt = token;
+    }
```

Comment:
suggestion: just to be absolutely sure that if a user is using js we have the right domain I would suggest the following changes:
```ts
if (!["admin", "store"].includes(domain)) {
  throw new Error(`'domain' must be wither 'admin' or 'store' received ${domain}`)
}

if (domain === "admin" {
  this.adminJwt = token
  return
}

this.storeJwt = token
```
wdyt?

Reply:
I agree. I think it would be good to implement a check during runtime as well as on the type level.

Reply:
added for this function as well as for getJwt() function